tests/console/sshd.pm

```diff
@@ -50,31 +50,39 @@ sub run {
     assert_script_run("echo $changepwd | chpasswd");
 
     opensusebasetest::select_serial_terminal();
-
-    # Make interactive SSH connection as the new user
-    type_string "ssh -v -l $ssh_testman localhost -t\n";
-    wait_serial('Are you sure you want to continue connecting (yes/no)?', undef, 0, no_regex => 1);
-    type_string "yes\n";
-    wait_serial('Password:', undef, 0, no_regex => 1);
-    type_string "$ssh_testman_passwd\n";
-    wait_serial('sshboy@susetest:~>', undef, 0, no_regex => 1);
-    type_string "export PS1='# '\n";
-
-    # Check that we are really in the SSH session
-    assert_script_run 'echo $SSH_TTY | grep "\/dev\/pts\/"';
-    assert_script_run 'ps ux | egrep ".* \? .* sshd\:"';
-    assert_script_run "whoami | grep $ssh_testman";
-    assert_script_run "mkdir .ssh";
-
-    # Exit properly and check we're root again
-    script_run("exit", 0);
-    assert_script_run "whoami | grep root";
+    if (is_serial_terminal()) {
```

Comment:
I consider this huge if/else too complex and not easy to read. The tests should be written so that they are easy to read from a testers point of view and not unfold into two completely different branches of test flow. Can we please try to find a nicer way?

@OleksandrOrlov this looks related to what you would like to prevent as well, doesn't it?

Reply:
Hello @okurz, I don't really see any other option over here, because of when the VirtOS serial terminal is available we wanna run this interactive part and when it isn't we wanna run the other.

Reply:
What is so complex on this if/else ? Do you know what complex even mean ?

Reply:
@okurz tbh the if/else looks fine (from a condition perspective) however while the instructions are long, I think they are more explicit... which helps.